File: data-structures-algorithms/searching/binary-search/binarySearch.ts
function binarySearch(hayStack: number[], needle: number): boolean {
  let low = 0;
  let high = hayStack.length;
  for (let i = 0; i < hayStack.length; i++) {
    if (low > high) break;
    const mid = Math.floor(low + (high - low) / 2);
    const value = hayStack[mid];
    if (value === needle) return true;
    else if (value > needle) high = mid;
    else low = mid + 1;
  }
  return false;
}

console.log(binarySearch([2, 5, 7, 8, 9, 10, 14, 18, 19], 3));

File: data-structures-algorithms/searching/binary-search/binary_search.py
import math

def binary_search(arr:list[int],target:int)->int:
    low = 0
    high = len(arr)
    while low <= high:
        mid = math.floor(low + (high-low) // 2)
        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            low = mid + 1
        else:
            high = mid
    return -1

print(binary_search([1,2,3,4,5,6,7,8,9,10], 8))